JSIL/AssemblyTranslator.cs

```diff
@@ -514,47 +528,47 @@ struct MethodToAnalyze {
 
             Action<int> writeAssembly = (i) => {
                 var assembly = assemblies[i];
+                bool stubbed = IsStubbed(assembly);
+
+                foreach (var pair in results) {
+                    long existingSize;
+                    foreach (var emmitterFactory in pair.EmitterGroup.MakeAssemblyEmitterFactory(this, assembly)) {
+                        var outputPath = FormatOutputFilename(emmitterFactory.AssemblyPathAndFilename);
+                        if (!Manifest.GetExistingSize(assembly, emmitterFactory.Id, out existingSize)) {
+                            using (var outputStream = new MemoryStream(DefaultStreamCapacity)) {
+                                var sourceMapBuilder = Configuration.BuildSourceMap.GetValueOrDefault() ? new SourceMapBuilder() : null;
+                                var context = MakeDecompilerContext(assembly.MainModule);
+
+                                try {
+                                    var tw = new StreamWriter(outputStream, Encoding.ASCII);
+                                    var formatter = new JavascriptFormatter(tw, sourceMapBuilder, this.TypeInfoProvider, Manifest, assembly, Configuration, stubbed);
+                                    var emitter = emmitterFactory.MakeAssemblyEmitter(formatter);
+                                    TranslateSingleAssemblyInternal(emitter, context, assembly, outputStream, sourceMapBuilder);
```

Comment:
Hmm, you are right that most of the work happens before. Maybe a formal split (with a rename) is sufficient to make it work out okay. I'll make some time to review the code and see whether the split is correct or not.